### PR TITLE
fix: fix sass deprecation errors

### PR DESCRIPTION
**Pull request by HeartSquared**
## Why

[KZN-2655](https://cultureamp.atlassian.net/jira/software/c/projects/KZN/boards/634?selectedIssue=KZN-2655)

Kaizen Sass deprecation messages are showing up in consumer repos 😬 

## What

Fix sass deprecation warnings

Also fixes:
- Button v1 right-to-left padding
- TextField reversed success icon styles

[KZN-2655]: https://cultureamp.atlassian.net/browse/KZN-2655?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/Checkbox/Checkbox/Checkbox.module.scss [hunk 1]
```diff
@@ -10,11 +10,11 @@ $dt-color-checkbox-background-color-checked: $color-gray-500;
 
 // override bootstrap styles
 .checkbox.checkbox {
-  @include form-input-visually-hidden;
-
   width: $checkbox-size;
   height: $checkbox-size;
   margin: 0;
+
+  @include form-input-visually-hidden;
```
Comment (mcwinter07) [inline packages/components/src/Checkbox/Checkbox/Checkbox.module.scss:17]: Wow, we have some super old style rules in there if we're overriding bootstrap 😆 
Screenshot 2024-10-17 at 2 00 51 PM

Comment (mcwinter07) [reply packages/components/src/Checkbox/Checkbox/Checkbox.module.scss:17]: heheh, that is to say I don't think we need to 🪓  that in this PR

File: packages/components/src/Input/InputSearch/InputSearch.module.scss [hunk 2]
```diff
@@ -44,6 +38,12 @@ $classname--input: '.input[type="search"]';
   font-size: 1rem;
   line-height: 1.5;
 
+  @include form-input-reset;
```
Comment (mcwinter07) [inline packages/components/src/Input/InputSearch/InputSearch.module.scss:41]: Will the reset coming after line 38 and 39 mean the font-size and line-height are superseded by the mixin?

File: packages/components/src/TitleBlockZen/TitleBlockZen.module.scss [hunk 3]
```diff
@@ -360,30 +356,28 @@ $tab-container-height-medium-and-small-collapsed: 0;
...
   flex-shrink: 0;
-  width: 10 * $ca-grid;
+  width: var(--spacing-240);
+  margin-inline: var(--spacing-12) 0;
 }
 
 .pageSwitcherSelectUnderneathTitle {
   flex-shrink: 0;
-  max-width: 10 * $ca-grid;
+  max-width: var(--spacing-240);
```
Comment (mcwinter07) [inline packages/components/src/TitleBlockZen/TitleBlockZen.module.scss:380]: nice 😉  love to see the strange * ca-grid getting removed

File: packages/components/src/Checkbox/Checkbox/Checkbox.module.scss (hunk 1 above)
Comment (HeartSquared) [reply packages/components/src/Checkbox/Checkbox/Checkbox.module.scss:17]: Yeah we have some pretty odd things 😬 But best to remove it separately if we want to :)

File: packages/components/src/Input/InputSearch/InputSearch.module.scss (hunk 2 above)
Comment (HeartSquared) [reply packages/components/src/Input/InputSearch/InputSearch.module.scss:41]: Good catch! I'll move this one back. Not sure why I moved it to begin with 😬

**Review by mcwinter07 — APPROVED** [review]

**Pull request merged**